From: Danila Tikhonov <danila@jiaxyga.com>
Date: Thu, 16 May 2024 00:24:04 +0300
Subject: drm: panel: Generate tianma-nt36672c driver

Signed-off-by: Danila Tikhonov <danila@jiaxyga.com>
---
 drivers/gpu/drm/panel/Kconfig                      |  10 +
 drivers/gpu/drm/panel/Makefile                     |   1 +
 ...-huaxing-nt36672c.c => panel-tianma-nt36672c.c} | 203 ++++++++++++++++-----
 3 files changed, 172 insertions(+), 42 deletions(-)
 copy drivers/gpu/drm/panel/{panel-huaxing-nt36672c.c => panel-tianma-nt36672c.c} (53%)

diff --git a/drivers/gpu/drm/panel/Kconfig b/drivers/gpu/drm/panel/Kconfig
index 94d6bdb..8a6c673 100644
--- a/drivers/gpu/drm/panel/Kconfig
+++ b/drivers/gpu/drm/panel/Kconfig
@@ -1066,6 +1066,16 @@ config DRM_PANEL_TDO_TL070WSH30
 	  24 bit RGB per pixel. It provides a MIPI DSI interface to
 	  the host, a built-in LED backlight and touch controller.
 
+config DRM_PANEL_TIANMA_NT36672C
+	tristate "Tianma FHD panel with NT36672C controller"
+	depends on OF
+	depends on DRM_MIPI_DSI
+	depends on BACKLIGHT_CLASS_DEVICE
+	select VIDEOMODE_HELPERS
+	help
+	  Say Y or M here if you want to enable support for the Tianma FHD
+	  (2400x1080@120Hz) video mode panel.
+
 config DRM_PANEL_TPO_TD028TTEC1
 	tristate "Toppoly (TPO) TD028TTEC1 panel driver"
 	depends on OF && SPI
diff --git a/drivers/gpu/drm/panel/Makefile b/drivers/gpu/drm/panel/Makefile
index b8c1260..cc482d3 100644
--- a/drivers/gpu/drm/panel/Makefile
+++ b/drivers/gpu/drm/panel/Makefile
@@ -106,6 +106,7 @@ obj-$(CONFIG_DRM_PANEL_SONY_TD4353_JDI) += panel-sony-td4353-jdi.o
 obj-$(CONFIG_DRM_PANEL_SONY_TULIP_TRULY_NT35521) += panel-sony-tulip-truly-nt35521.o
 obj-$(CONFIG_DRM_PANEL_STARTEK_KD070FHFID015) += panel-startek-kd070fhfid015.o
 obj-$(CONFIG_DRM_PANEL_TDO_TL070WSH30) += panel-tdo-tl070wsh30.o
+obj-$(CONFIG_DRM_PANEL_TIANMA_NT36672C) += panel-tianma-nt36672c.o
 obj-$(CONFIG_DRM_PANEL_TPO_TD028TTEC1) += panel-tpo-td028ttec1.o
 obj-$(CONFIG_DRM_PANEL_TPO_TD043MTEA1) += panel-tpo-td043mtea1.o
 obj-$(CONFIG_DRM_PANEL_TPO_TPG110) += panel-tpo-tpg110.o
diff --git a/drivers/gpu/drm/panel/panel-huaxing-nt36672c.c b/drivers/gpu/drm/panel/panel-tianma-nt36672c.c
similarity index 53%
copy from drivers/gpu/drm/panel/panel-huaxing-nt36672c.c
copy to drivers/gpu/drm/panel/panel-tianma-nt36672c.c
index 428ee60..29fc663 100644
--- a/drivers/gpu/drm/panel/panel-huaxing-nt36672c.c
+++ b/drivers/gpu/drm/panel/panel-tianma-nt36672c.c
@@ -18,7 +18,7 @@
 #include <drm/drm_panel.h>
 #include <drm/drm_probe_helper.h>
 
-struct nt36672c_huaxing {
+struct nt36672c_tianma {
 	struct drm_panel panel;
 	struct mipi_dsi_device *dsi;
 	struct drm_dsc_config dsc;
@@ -27,12 +27,12 @@ struct nt36672c_huaxing {
 };
 
 static inline
-struct nt36672c_huaxing *to_nt36672c_huaxing(struct drm_panel *panel)
+struct nt36672c_tianma *to_nt36672c_tianma(struct drm_panel *panel)
 {
-	return container_of(panel, struct nt36672c_huaxing, panel);
+	return container_of(panel, struct nt36672c_tianma, panel);
 }
 
-static void nt36672c_huaxing_reset(struct nt36672c_huaxing *ctx)
+static void nt36672c_tianma_reset(struct nt36672c_tianma *ctx)
 {
 	gpiod_set_value_cansleep(ctx->reset_gpio, 0);
 	usleep_range(5000, 6000);
@@ -42,25 +42,142 @@ static void nt36672c_huaxing_reset(struct nt36672c_huaxing *ctx)
 	usleep_range(10000, 11000);
 }
 
-static int nt36672c_huaxing_on(struct nt36672c_huaxing *ctx)
+static int nt36672c_tianma_on(struct nt36672c_tianma *ctx)
 {
 	struct mipi_dsi_multi_context dsi_ctx = { .dsi = ctx->dsi };
 
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0x10);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
-	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xb0, 0x00);
-	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0xe0);
+	mipi_dsi_generic_write_seq_multi(&dsi_ctx, 0xc2, 0x1b, 0xa0);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0x20);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
-	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x35, 0x82);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x06, 0x64);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x69, 0xd0);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xf2, 0x65);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xf3, 0x54);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xf4, 0x65);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xf5, 0x54);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xf6, 0x65);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xf7, 0x54);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xf8, 0x65);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xf9, 0x54);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0x24);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x02, 0x2f);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x03, 0x2d);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x04, 0x2e);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x05, 0x2c);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x06, 0x8b);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x07, 0x8c);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x1a, 0x2f);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x1b, 0x2d);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x1c, 0x2e);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x1d, 0x2c);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x1e, 0x8b);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x1f, 0x8c);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x33, 0x03);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x34, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x4d, 0x05);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x4e, 0x33);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x4f, 0x33);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, MIPI_DCS_WRITE_CONTROL_DISPLAY,
+				     0x33);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x79, 0x03);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x7a, 0x03);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x7b, 0x8d);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x7d, 0x05);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x80, 0x05);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x81, 0x05);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xa0, 0x0d);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, MIPI_DCS_READ_PPS_START, 0x0d);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xa3, 0x03);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xa4, 0x06);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xa5, 0x06);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xe9, 0x02);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0x25);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x21, 0x00);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x66, 0x40);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x67, 0x29);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x79, 0x7a);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xda, 0x00);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xe0, 0x00);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xf0, 0x00);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xf1, 0x44);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0x26);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x01, 0xea);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x04, 0xea);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x05, 0x08);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x06, 0x0f);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x08, 0x0f);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x81, 0x0d);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x83, 0x03);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x88, 0x03);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0x27);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x6e, 0x23);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x6f, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x73, 0x21);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x74, 0x03);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x7e, 0x63);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x83, 0x63);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x88, 0x03);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x89, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0x2a);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x00, 0x91);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x03, 0x20);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x07, 0x56);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x0a, 0x70);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x0c, 0x09);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x0d, 0x40);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x0e, 0x02);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x0f, 0x00);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x11, 0xf0);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x15, 0x0e);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x16, 0xa8);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x19, 0x0e);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x1a, 0x7c);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x1e, 0x38);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x1f, 0x38);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x20, 0x38);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x27, 0x80);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x28, 0xb2);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x29, 0x1d);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x2a, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x2b, 0x00);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, MIPI_DCS_WRITE_LUT, 0x04);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x2f, 0x06);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, MIPI_DCS_SET_PARTIAL_ROWS, 0x1e);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, MIPI_DCS_SET_PARTIAL_COLUMNS,
+				     0x42);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x33, 0xb0);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x34, 0xb4);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x35, 0x34);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, MIPI_DCS_SET_ADDRESS_MODE, 0xc4);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x37, 0xaf);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x38, 0x38);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x39, 0xc0);
+	mipi_dsi_dcs_set_pixel_format_multi(&dsi_ctx, 0x1e);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0x2b);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xb7, 0x06);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xb8, 0x13);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xc0, 0x03);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0xf0);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x1c, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x33, 0x01);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x5a, 0x00);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x9c, 0x00);
-	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xbe, 0x08);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0xc0);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x9c, 0x11);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x9d, 0x11);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0xe0);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x35, 0x82);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0x10);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
 	mipi_dsi_dcs_set_display_brightness_multi(&dsi_ctx, 0x00c1);
@@ -139,11 +256,13 @@ static int nt36672c_huaxing_on(struct nt36672c_huaxing *ctx)
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x3f, 0x01);
 	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0x43, 0x08);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xff, 0x10);
+	mipi_dsi_dcs_write_seq_multi(&dsi_ctx, 0xfb, 0x01);
 
 	return dsi_ctx.accum_err;
 }
 
-static int nt36672c_huaxing_off(struct nt36672c_huaxing *ctx)
+static int nt36672c_tianma_off(struct nt36672c_tianma *ctx)
 {
 	struct mipi_dsi_multi_context dsi_ctx = { .dsi = ctx->dsi };
 
@@ -157,9 +276,9 @@ static int nt36672c_huaxing_off(struct nt36672c_huaxing *ctx)
 	return dsi_ctx.accum_err;
 }
 
-static int nt36672c_huaxing_prepare(struct drm_panel *panel)
+static int nt36672c_tianma_prepare(struct drm_panel *panel)
 {
-	struct nt36672c_huaxing *ctx = to_nt36672c_huaxing(panel);
+	struct nt36672c_tianma *ctx = to_nt36672c_tianma(panel);
 	struct device *dev = &ctx->dsi->dev;
 	struct drm_dsc_picture_parameter_set pps;
 	int ret;
@@ -170,9 +289,9 @@ static int nt36672c_huaxing_prepare(struct drm_panel *panel)
 		return ret;
 	}
 
-	nt36672c_huaxing_reset(ctx);
+	nt36672c_tianma_reset(ctx);
 
-	ret = nt36672c_huaxing_on(ctx);
+	ret = nt36672c_tianma_on(ctx);
 	if (ret < 0) {
 		dev_err(dev, "Failed to initialize panel: %d\n", ret);
 		gpiod_set_value_cansleep(ctx->reset_gpio, 1);
@@ -199,13 +318,13 @@ static int nt36672c_huaxing_prepare(struct drm_panel *panel)
 	return 0;
 }
 
-static int nt36672c_huaxing_unprepare(struct drm_panel *panel)
+static int nt36672c_tianma_unprepare(struct drm_panel *panel)
 {
-	struct nt36672c_huaxing *ctx = to_nt36672c_huaxing(panel);
+	struct nt36672c_tianma *ctx = to_nt36672c_tianma(panel);
 	struct device *dev = &ctx->dsi->dev;
 	int ret;
 
-	ret = nt36672c_huaxing_off(ctx);
+	ret = nt36672c_tianma_off(ctx);
 	if (ret < 0)
 		dev_err(dev, "Failed to un-initialize panel: %d\n", ret);
 
@@ -215,7 +334,7 @@ static int nt36672c_huaxing_unprepare(struct drm_panel *panel)
 	return 0;
 }
 
-static const struct drm_display_mode nt36672c_huaxing_mode = {
+static const struct drm_display_mode nt36672c_tianma_mode = {
 	.clock = (1080 + 73 + 12 + 40) * (2400 + 32 + 2 + 30) * 120 / 1000,
 	.hdisplay = 1080,
 	.hsync_start = 1080 + 73,
@@ -230,26 +349,26 @@ static const struct drm_display_mode nt36672c_huaxing_mode = {
 	.type = DRM_MODE_TYPE_DRIVER,
 };
 
-static int nt36672c_huaxing_get_modes(struct drm_panel *panel,
-				      struct drm_connector *connector)
+static int nt36672c_tianma_get_modes(struct drm_panel *panel,
+				     struct drm_connector *connector)
 {
-	return drm_connector_helper_get_modes_fixed(connector, &nt36672c_huaxing_mode);
+	return drm_connector_helper_get_modes_fixed(connector, &nt36672c_tianma_mode);
 }
 
-static const struct drm_panel_funcs nt36672c_huaxing_panel_funcs = {
-	.prepare = nt36672c_huaxing_prepare,
-	.unprepare = nt36672c_huaxing_unprepare,
-	.get_modes = nt36672c_huaxing_get_modes,
+static const struct drm_panel_funcs nt36672c_tianma_panel_funcs = {
+	.prepare = nt36672c_tianma_prepare,
+	.unprepare = nt36672c_tianma_unprepare,
+	.get_modes = nt36672c_tianma_get_modes,
 };
 
-static int nt36672c_huaxing_probe(struct mipi_dsi_device *dsi)
+static int nt36672c_tianma_probe(struct mipi_dsi_device *dsi)
 {
 	struct device *dev = &dsi->dev;
-	struct nt36672c_huaxing *ctx;
+	struct nt36672c_tianma *ctx;
 	int ret;
 
-	ctx = devm_drm_panel_alloc(dev, struct nt36672c_huaxing, panel,
-				   &nt36672c_huaxing_panel_funcs,
+	ctx = devm_drm_panel_alloc(dev, struct nt36672c_tianma, panel,
+				   &nt36672c_tianma_panel_funcs,
 				   DRM_MODE_CONNECTOR_DSI);
 	if (IS_ERR(ctx))
 		return PTR_ERR(ctx);
@@ -286,7 +405,7 @@ static int nt36672c_huaxing_probe(struct mipi_dsi_device *dsi)
 	ctx->dsc.dsc_version_major = 1;
 	ctx->dsc.dsc_version_minor = 1;
 
-	/* TODO: Pass slice_per_pkt = 2 */
+	/* TODO: Pass slice_per_pkt = 1 */
 	ctx->dsc.slice_height = 20;
 	ctx->dsc.slice_width = 540;
 	/*
@@ -308,9 +427,9 @@ static int nt36672c_huaxing_probe(struct mipi_dsi_device *dsi)
 	return 0;
 }
 
-static void nt36672c_huaxing_remove(struct mipi_dsi_device *dsi)
+static void nt36672c_tianma_remove(struct mipi_dsi_device *dsi)
 {
-	struct nt36672c_huaxing *ctx = mipi_dsi_get_drvdata(dsi);
+	struct nt36672c_tianma *ctx = mipi_dsi_get_drvdata(dsi);
 	int ret;
 
 	ret = mipi_dsi_detach(dsi);
@@ -320,22 +439,22 @@ static void nt36672c_huaxing_remove(struct mipi_dsi_device *dsi)
 	drm_panel_remove(&ctx->panel);
 }
 
-static const struct of_device_id nt36672c_huaxing_of_match[] = {
-	{ .compatible = "huaxing,nt36672c" }, // FIXME
+static const struct of_device_id nt36672c_tianma_of_match[] = {
+	{ .compatible = "tianma,nt36672c" }, // FIXME
 	{ /* sentinel */ }
 };
-MODULE_DEVICE_TABLE(of, nt36672c_huaxing_of_match);
+MODULE_DEVICE_TABLE(of, nt36672c_tianma_of_match);
 
-static struct mipi_dsi_driver nt36672c_huaxing_driver = {
-	.probe = nt36672c_huaxing_probe,
-	.remove = nt36672c_huaxing_remove,
+static struct mipi_dsi_driver nt36672c_tianma_driver = {
+	.probe = nt36672c_tianma_probe,
+	.remove = nt36672c_tianma_remove,
 	.driver = {
-		.name = "panel-nt36672c-huaxing",
-		.of_match_table = nt36672c_huaxing_of_match,
+		.name = "panel-nt36672c-tianma",
+		.of_match_table = nt36672c_tianma_of_match,
 	},
 };
-module_mipi_dsi_driver(nt36672c_huaxing_driver);
+module_mipi_dsi_driver(nt36672c_tianma_driver);
 
 MODULE_AUTHOR("linux-mdss-dsi-panel-driver-generator <fix@me>"); // FIXME
-MODULE_DESCRIPTION("DRM driver for nt36672c huaxing fhd video mode dsi panel");
+MODULE_DESCRIPTION("DRM driver for nt36672c tianma fhd video mode dsi panel");
 MODULE_LICENSE("GPL");
